### PR TITLE
Use ES256 instead of RS256 to generate shorter examples

### PR DESCRIPTION
**Pull request by danielfett**

### CODE REVIEW

File: sd_jwt/operations.py [hunk 1]
```diff
@@ -231,7 +233,7 @@ def _create_signed_jwt(self):
 
         # Sign the SD-JWT using the issuer's key
         self.sd_jwt = JWS(payload=dumps(self.sd_jwt_payload))
-        _headers = {"alg": self._sign_alg, "kid": self._issuer_key.thumbprint()}
+        _headers = {"alg": self._sign_alg}
```
Comment (danielfett) [inline sd_jwt/operations.py:236]: @bc-pi I removed the "kid" header parameter from the examples. It used to contain the key thumbprint. AFAIK there is no requirement for a "kid" header parameter and just always putting the thumbprint there is not the right solution anyway - any objections?

Comment (danielfett) [reply sd_jwt/operations.py:236]: (I'm not sure why `kid` was in there in the first place.)

Comment (peppelinux) [reply sd_jwt/operations.py:236]: It was me @danielfett, it's not relevant for the draft and its examples, please go ahead :-)

Comment (bc-pi) [reply sd_jwt/operations.py:236]: no objections

File: examples/simple/sd_jwt_payload.json [hunk 2]
```diff
@@ -14,14 +14,13 @@
...
-        659kg4hVo9dbGoCJE3ZGF_eaetE30UhBUEgpGwrDrQiJ9zqprmcFfr3qvvkG
-        jtth8Zgl1eM2bJcOwE7PCBHWTKWYs152R7g6Jg2OVph-a8rq-q79MhKG5QoW
-        _mTz10QT_6H4c7PjWG1fjh8hpWNnbP_pv6d1zSwZfc5fl6yVRL0DV0V3lGHK
-        e2Wqf_eNGjBrBLVklDTk8-stX_MWLcR-EGmXAOv0UBWitS_dXJKJu-vXJyw1
-        4nHSGuxTIK2hx1pttMft9CsvqimXKeDTU14qQL1eE7ihcw",
-      "e": "AQAB"
+      "kty": "EC",
+      "use": "sig",
+      "alg": "ES256",
+      "kid": "2",
```
Comment (bc-pi) [inline examples/simple/sd_jwt_payload.json:20]: I don't think `use`, `alg` and `kid` should be included here.

File: examples/simple/verified_contents.json [hunk 3]
```diff
@@ -5,14 +5,13 @@
...
-        659kg4hVo9dbGoCJE3ZGF_eaetE30UhBUEgpGwrDrQiJ9zqprmcFfr3qvvkG
-        jtth8Zgl1eM2bJcOwE7PCBHWTKWYs152R7g6Jg2OVph-a8rq-q79MhKG5QoW
-        _mTz10QT_6H4c7PjWG1fjh8hpWNnbP_pv6d1zSwZfc5fl6yVRL0DV0V3lGHK
-        e2Wqf_eNGjBrBLVklDTk8-stX_MWLcR-EGmXAOv0UBWitS_dXJKJu-vXJyw1
-        4nHSGuxTIK2hx1pttMft9CsvqimXKeDTU14qQL1eE7ihcw",
-      "e": "AQAB"
+      "kty": "EC",
+      "use": "sig",
+      "alg": "ES256",
+      "kid": "2",
```
Comment (bc-pi) [inline examples/simple/verified_contents.json:11]: not really meaningful or useful in this context (I don't think?) but take up space and potentially distract or confuse

File: sd_jwt/demo_settings.py [hunk 4]
```diff
@@ -1,31 +1,27 @@
...
 
+HOLDER_KEY_DICT = {
+    "kty": "EC",
+    "d": "5K5SCos8zf9zRemGGUl6yfok-_NiiryNZsvANWMhF-I",
+    "use": "sig",
+    "crv": "P-256",
+    "kid": "2",
+    "x": "TCAER19Zvu3OHF4j4W4vfSVoHIP1ILilDls7vCeGemc",
+    "y": "ZxjiWWbZMQGHVWKVQ4hbSIirsVfuecCE6t4jT9F2HZQ",
+    "alg": "ES256"
```
Comment (bc-pi) [inline sd_jwt/demo_settings.py:26]: I'm gussing this is where they come from but, as per my previous comments, I don't think use/kid/alg should be in the holder confirmation key.

File: examples/simple/verified_contents.json (hunk 3 above)
Comment (bc-pi) [reply examples/simple/verified_contents.json:11]: use alg and kid

**Review by bc-pi — CHANGES_REQUESTED** [review]
Looks pretty good overall! I even double checked some of the signatures with my JOSE/JWT library. 

I do, however, think that before merging a few parameters should be removed from the holder jwk.

File: examples/simple/verified_contents.json (hunk 3 above)
Comment (danielfett) [reply examples/simple/verified_contents.json]: You're right! I removed them.

File: sd_jwt/demo_settings.py (hunk 4 above)
Comment (danielfett) [reply sd_jwt/demo_settings.py]: As above - removed the three superfluous elements.

**Review by bc-pi — APPROVED** [review]

**Comment by bc-pi** [comment]
Approved!

**Review by Sakurann — APPROVED** [review]

**Pull request merged**